crates/agglayer-config/src/log.rs

```diff
@@ -67,7 +67,7 @@ impl From<LogLevel> for EnvFilter {
 /// specify the output location as a string, which is then parsed into the
 /// appropriate enum variant. If the string is not recognized to be either
 /// `stdout` or `stderr`, it is assumed to be a file path.
-#[derive(Debug, Clone, Default)]
+#[derive(Serialize, Debug, Clone, Default, PartialEq, Eq)]
```

Comment:
I see this one has a custom `Deserialize` impl but derived `Serialize` impl. These are not consistent with each other, i.e. the serialize/deserialize roundtrip will not work as expected in some cases.

Reply:
The `Serialize` derive is ok because the default value is an empty hashmap

Reply:
> The `Serialize` derive is ok because the default value is an empty hashmap

But the serialize impl supports writing out non-default values too, at least in principle. That might cause issues if something like "dump current config" functionality is implemented.

Reply:
We'll not do any `dump current config` functionality for now and this part will certainly be removed as it is fairly useless.

Reply:
Understood. I guess I just don't like having inconsistent impls around in the code.

I think it could be addressed by adding the `rename_all` attribute as with the other stuff and annotating the `File` arm with `#[serde(untagged)]`. That should remove the need for the custom impl completely, unless I am missing some other issue.